go/toolchains.rst

```diff
@@ -18,22 +27,568 @@ customize the behavior of the core_ go rules.
+| This is the host platform tuple.                                                                 |
+| If it is not set, it defaults to the same as target.                                             |
+| If it is set to a different value to target, then this is declaring a cross-compiling toolchain. |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`sdk`                   | :type:`string`              | |mandatory|                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| This is the name of the SDK to use for this toolchain.                                           |
+| The SDK must have been registered using go_sdk_.                                                 |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`constraints`           | :type:`label_list`          | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| This list is added to the host and or target constraints when declaring the toolchains.          |
+| It allows the declaration f additional constraints that must be matched for the toolchain to     |
+| be automatically selected.                                                                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`link_flags`            | :type:`string_list`         | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| The link flags are directly exposed on the toolchain.                                            |
+| They can be used to specify target specific flags that go linking actions should apply when      |
+| using this toolchain.                                                                            |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`cgo_link_flags`        | :type:`string_list`         | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| The cgo link flags are directly exposed on the toolchain.                                        |
+| They can be used to specify target specific flags that c linking actions generated by cgo        |
+| should apply when using this toolchain.                                                          |
++--------------------------------+-----------------------------+-----------------------------------+
+
+The toolchain object
+~~~~~~~~~~~~~~~~~~~~
+
+When you get a go toolchain from a context (see use_) it exposes a number of fields.
+
+The public interface is the actions member.
```

Comment:
Instead of listing the emitting functions at the bottom of this section, I think it would be better to have a hierarchical list for the whole object here.

* go_toolchain
  * actions
    * asm
    * ...